### PR TITLE
Sync CA Capture and Refund with OMS

### PR DESCRIPTION
**Pull request by shubhamk67**
## Summary
Describe the changes proposed in this pull request:
- What is the motivation for this change?
Currently, captures and refunds initiated from CA are not reflected in OMS.
- What existing problem does this pull request solve?
Creating a gateway log in OMS when captures or refunds are completed from CA. This will provide better visibility for merchants.

### CODE REVIEW

**Review by amihajlovski — DISMISSED** [review]

File: force-app/main/default/classes/NonPaymentWebhookHandler.cls [hunk 1]
```diff
@@ -23,7 +23,9 @@ global without sharing class NonPaymentWebhookHandler {
                 return responseWithReason(NO_PAYMENT_FOUND);
             }
             processNotification(paymentAuthorization, notificationRequestItem);
-            createGatewayLog(paymentAuthorization, notificationRequestItem, requestBody);
+            if(requiresGatewayLog(notificationRequestItem)) {
```
Comment (dcardos) [inline force-app/main/default/classes/NonPaymentWebhookHandler.cls:26]: For my OCD, if you can add a space after the "if" would be great. haha

File: force-app/main/default/classes/NonPaymentWebhookHandler.cls [hunk 2]
```diff
@@ -103,4 +105,40 @@ global without sharing class NonPaymentWebhookHandler {
...
+        List<Refund> refunds = [
+            SELECT Id
+            FROM Refund
+            WHERE GatewayRefNumber = :gatewayReference 
+            AND OrderPaymentSummaryId != NULL
+        ];
+        return !refunds.isEmpty();
+    }
+
+    private static Boolean requiresGatewayLog(NotificationRequestItem requestItem) {
```
Comment (dcardos) [inline force-app/main/default/classes/NonPaymentWebhookHandler.cls:129]: Really nice method, easy to read!

**Review by dcardos — CHANGES_REQUESTED** [review]
Neat code, but as discussed on mattermost, there is only one use case missing: refund when it is a SALE flow: no PA present, only a payment record and a refund is triggered in CA only.

**Review by dcardos — APPROVED** [review]
Great work!

**Pull request merged**